main.go

```diff
@@ -0,0 +1,182 @@
+	fromLatestTag     = flag.Bool("from-latest-tag", false, "If you use tags, set to true to get changes from latest tag.")
+	fromCommit        = flag.StringP("from", "f", "", "The beginning commit. Defaults to beginning of the repository history")
+	toCommit          = flag.StringP("to", "t", "HEAD", "The last commit.")
+	sinceTime         = flag.String("since", "", "Show commits more recent than a specific date. Use RFC3339 time '2017-08-01T00:00:00Z'. Takes precedence over to/from.")
+	untilTime         = flag.String("until", "", "Show commits older than a specific date. Defaults to current time if not set, but --since is. Takes precedence over to/from.")
+	version           = flag.StringP("version", "v", "", "The version you are creating")
+	repoLink          = flag.StringP("repo", "r", "", "The repository URL. Defaults to `$(git remote get-url origin)` if using a local provider")
+	includeAllCommits = flag.Bool("include-all", false, "Set to true to include all commits in the changelog. Commit messages that cannot be parsed will be placed in a section titled \"Unknown\".")
+
+	provider = flag.StringP("provider", "p", "local", fmt.Sprintf(`The provider to use. Must be one of %s`, strings.Join(providers, ", ")))
+
+	token = flag.String("token", "", "API token for remote provider. Only applies to github provider")
+
+	gitDir   = flag.String("git-dir", "", "The path to the git directory. If no '--repo' is set, defaults to `$(pwd)/.git`. Only applies to local provider")
+	workTree = flag.String("work-tree", "", "The path to the directory containing the .git directory. Only applies to local provider.")
+
+	// outfile = flagSet.String("outfile", "", "") // TODO to maintain clog compatibility
+	// infile = flagSet.String("infile", "", "") // TODO to maintain clog compatibility
+	// outputFormat = flagSet.String("output-format", "markdown", "markdown or json") // TODO to maintain clog compatibility
+
+	// debug = flagSet.Bool("debug", false, "Set to output debug logging")
+)
+
+func validateProvider(provider string) error {
+	for i := range providers {
+		if provider == providers[i] {
+			return nil
+		}
+	}
+	return fmt.Errorf("Provider %s not found! Must be one of %s", provider, strings.Join(providers, ", "))
+}
+
+func init() {
+	viper.SetEnvPrefix("changelog")
+	viper.AutomaticEnv()
+	viper.SetConfigName(".clog")
+	viper.SetConfigType("toml")
+	viper.BindPFlags(flag.CommandLine)
+	viper.ReadInConfig()
+}
+
+func exitOnError(err error) {
+	fmt.Printf("Fatal Error: %s", err.Error())
+	os.Exit(1)
+}
+
+func main() {
+	flag.Parse()
+
+	if err := validateProvider(viper.GetString("provider")); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	var style linkStyle.Style
+	var querier changelog.Querier
+
+	switch viper.GetString("provider") {
+	case "github":
+		style = linkStyle.Github
+		querier = changelog.NewGithubQuerier(viper.GetString("repo"), viper.GetString("token"))
+	case "stash", "bitbucket":
+		style = linkStyle.Stash
```

Comment:
It might be good to go ahead and remove this since it won't even get past the validation anyway.

Reply:
Yea, I'll drop the gitlab one too